includes/chart.js

```diff
@@ -0,0 +1,330 @@
+                        var bbox = newNode.getBoundingClientRect();
+                        var deltaX = event.clientX - offsetX - bbox.x;
+                        var deltaY = event.clientY - offsetY - bbox.y;
+                        // Update polygon points
+                        var points = newNode.points;
+                        var numPoints = points.numberOfItems;
+                        for (var i = 0; i < numPoints; i++) {
+                            points.getItem(i).x += deltaX;
+                            points.getItem(i).y += deltaY;
+                        }
+                        // Update text position for polygon
+                        text.setAttribute("x", parseFloat(text.getAttribute("x")) + deltaX);
+                        text.setAttribute("y", parseFloat(text.getAttribute("y")) + deltaY);
+
+                    } else if (newNode.tagName === 'rect') {
+                        newNode.setAttribute("x", event.clientX - offsetX);
+                        newNode.setAttribute("y", event.clientY - offsetY);
+                        // Update text position for rectangle
+                        text.setAttribute("x", parseFloat(newNode.getAttribute("x")) + parseFloat(newNode.getAttribute("width")) / 2);
+                        text.setAttribute("y", parseFloat(newNode.getAttribute("y")) + parseFloat(newNode.getAttribute("height")) / 2);
+                    }
+
+                    updateEdges(row.id);
+                }
+
+                function stopMove() {
+                    document.removeEventListener('mousemove', moveNode);
+                    document.removeEventListener('mouseup', stopMove);
+                }
+
+                document.addEventListener('mousemove', moveNode);
+                document.addEventListener('mouseup', stopMove);
+            });
+
+        });
+
+        updateEdges(row.id);
+    });
+}
+
+
+// Function to update edge positions
+function updateEdges(rowID) {
+    row_template.links.forEach(function (link) {
+        var sourceNode = document.getElementById(`${rowID}_${link.source}`);
+        var targetNode = document.getElementById(`${rowID}_${link.target}`);
+
+        if (!sourceNode || !targetNode) {
+            return;
+        }
+
+        var sourceX, sourceY, targetX, targetY;
+
+        if (sourceNode.tagName === 'ellipse') {
+            sourceX = parseFloat(sourceNode.getAttribute("cx"));
+            sourceY = parseFloat(sourceNode.getAttribute("cy"));
+        } else if (sourceNode.tagName === 'polygon') {
+            // Get the bounding box of the polygon
+            var sourceBBox = sourceNode.getBoundingClientRect();
+            // Define an offset to adjust the initial edge position (empirical adjustment)
+            var offsetX = 275; // Adjust this value as needed (!!! This is a little temporary hack !!!)
+            var offsetY = 225; // Adjust this value as needed (!!! This is a little temporary hack !!!)
```

Comment:
You won't find these lines anymore after a few weeks. ;-) Please use variables for this instead of hard-coding.

Reply:
Deferred to #25